lib/libutee/tee_api_operations.c

```diff
@@ -1364,17 +1352,15 @@ void TEE_AEUpdateAAD(TEE_OperationHandle operation, const void *AADdata,
 TEE_Result TEE_AEUpdate(TEE_OperationHandle operation, const void *srcData,
 			uint32_t srcLen, void *destData, uint32_t *destLen)
 {
-	TEE_Result res;
-	size_t req_dlen;
-	uint64_t dl;
+	TEE_Result res = TEE_SUCCESS;
+	size_t req_dlen = 0;
+	uint64_t dl = 0;
 
-	if (operation == TEE_HANDLE_NULL ||
-	    (srcData == NULL && srcLen != 0) ||
-	    destLen == NULL ||
-	    (destData == NULL && *destLen != 0)) {
+	if (operation == TEE_HANDLE_NULL || (!srcData && srcLen)) {
 		res = TEE_ERROR_BAD_PARAMETERS;
 		goto out;
 	}
+	__utee_check_inout_annotation(destLen, sizeof(*destLen));
```

Comment:
Here also `__utee_check_outbuf_annotation(destData, destLen);` seems more appropriate.

Reply:
I'd prefer keeping this as is. But let's discuss it if you disagree.

Reply:
For what it worth, this seems totally acceptable to me -- the syscall doing what is strictly necessary, and the additional GP constraints checked in the GP layer.

Reply:
Makes sense. thanks for the feedback. 
I'll delete the several related review comments of mine so that they don't pollute more this review.

Reply:
Thanks